extensions/amp-video/0.1/amp-video.js

```diff
@@ -361,6 +361,12 @@ class AmpVideo extends AMP.BaseElement {
       return;
     }
 
+    // Resolve layoutCallback right away if the video is within a story, so it
+    // can be handled by the media pool as soon as possible.
+    if (this.isManagedByPool_()) {
```

Comment:
We had a bunch of bugs related to the media pool picking up a video element before it is fully initialized. How is that handled now?

E.g. if we are in prerender, then the sources would still change after this time.

Reply:
Good catch. Prerendering is the only case I can think of, so if we return a promise that resolves when the non cached sources are added we should be good?

Reply:
Yeah, something like `sourcesFullyInitializedPromise` which gets mixed into `promise` here, but for the mediaPool case you only return it

Reply:
PTAL